Guide for overriding custom assets

braem opened the pull request:
Closes #30 

Includes 
- An overall description
- Reason for existing (why to not just modify game files)
- File structure examples, correct and incorrect
- An example for overriding weapon scripts + image of default vs tf2 stickylauncher
- A description of the better performance of VPKs vs subfolders

![image](https://user-images.githubusercontent.com/9014762/83304593-f7fd1e00-a1b3-11ea-8942-2b1aee6f8608.png)
![image](https://user-images.githubusercontent.com/9014762/83304498-d00dba80-a1b3-11ea-9bcb-0f79d39a2e05.png)
![image](https://user-images.githubusercontent.com/9014762/83304518-da2fb900-a1b3-11ea-869d-c5a2d1c05758.png)
![image](https://user-images.githubusercontent.com/9014762/83304539-e61b7b00-a1b3-11ea-92b1-13f2cc4ba55a.png)
![image](https://user-images.githubusercontent.com/9014762/83304561-eddb1f80-a1b3-11ea-8a20-c78352dae439.png)

Gocnak commented:
Can you also do an example for HUD overriding? Like changing where the speedometer is, for example.

braem commented:
> Can you also do an example for HUD overriding? Like changing where the speedometer is, for example.

For sure! 🙂

braem commented:
Added a speedometer centering example:
![image](https://user-images.githubusercontent.com/9014762/83323537-5c4fca00-a214-11ea-8f40-5896b2312dff.png)
![image](https://user-images.githubusercontent.com/9014762/83323554-80aba680-a214-11ea-942d-01c7637e25f8.png)

Gocnak commented:
Oh I suppose something else to mention are the two commands `weapon_reload_scripts` and `weapon_reload_script_current`, for hotloading in the changes made to weapon scripts.

braem commented:
> Oh I suppose something else to mention are the two commands `weapon_reload_scripts` and `weapon_reload_script_current`, for hotloading in the changes made to weapon scripts.

Should these be added to the con commands list as well?

Gocnak commented:
Yes.

braem commented:
Added those weapon reload commands and also `hud_reloadcontrols` as to my knowledge this is specific to momentum, or at least not in TF2.

Added references to these in the guide.
